module/documents/item.mjs

```diff
@@ -1999,7 +1997,7 @@ export default class Item5e extends Item {
       throw new Error(`${type} advancement cannot be added to ${this.name}`);
     }
 
-    const advancement = new Advancement(this, data);
+    const advancement = new Advancement(data, {parent: this});
```

Comment:
I don't mind leaving the `Advancement`'s parent as the system data model because we are always using the `Advancement#item` getter anyways, so it doesn't add any confusion.

Reply:
That's true for reading, but for instantiating like we're doing here, it feels a little obtuse that you'd have to supply the `Item`s system data model instance as the `parent` rather than the `Item` itself. That's the main reason I suggested it. But maybe we don't expect consumers of this API to really instantiate `Advancement`s manually?